graph.py

```diff
@@ -0,0 +1,269 @@
+            else:
+                self.leftChild = Edge(city1, city2, distance)
+                return True
+        else:
+            if self.rightChild:
+                return self.rightChild.insert(city1, city2, distance)
+            else:
+                self.rightChild = Edge(city1, city2, distance)
+                return True
+
+class EdgeTree:
+    def __init__(self):
+        self.root = None
+        self.size = 0
+        self.sortedEdges = []
+
+    def insert(self, city1, city2, distance):
+        if self.root:
+            return self.root.insert(city1, city2, distance)
+        else:
+            self.root = Edge(city1, city2, distance)
+            return True
+
+    def DFS(self, root):
+        if not root:
+            return False
+        self.DFS(root.leftChild)
+        self.sortedEdges.append({"city1": root.city1, "city2" :root.city2,
+                                "distance" :root.distance})
+        print(root.distance)
+        self.DFS(root.rightChild)
+
+
+
+class Graph:
+    def __init__(self):
+        self.vertList = {}
+        self.numVertices = 0
+
+    def addVertex(self, continent, country, city, latitude, longitude):
+        self.numVertices = self.numVertices + 1
+        newVertex = Vertex(continent, country, city, latitude, longitude)
+        self.vertList[city] = newVertex
+        return newVertex
+
+    def getVertex(self, n):
+        if n in self.vertList:
+            return self.vertList[n]
+        else:
+            return None
+
+    def edgeExists(self, v1, v2):
+        if v1 in self.getVertex(v2).connectedTo.keys():
+            return True
+        else:
+            return False
+
+    def getDistance(self, city1, city2):
+
+        lat1 = self.vertList[city1].latitude
+        long1 = self.vertList[city1].longitude
+
+        lat2 = self.vertList[city2].latitude
+        long2 = self.vertList[city2].longitude
+
+        lat_dist = lat2 - lat1
+        long_dist = long2 - long1
+
+        return math.hypot(lat_dist, long_dist)
+
+    def __contains__(self,n):
+        return n in self.vertList
+
+    def addEdge(self, f, t, weight=0):
+        if f not in self.vertList:
+            nv = self.addVertex(f)
+        if t not in self.vertList:
+            nv = self.addVertex(t)
+        self.vertList[f].addNeighbor(self.vertList[t], weight)
+
+    def getVertices(self):
+        return self.vertList.keys()
+
+    def __iter__(self):
+        return iter(self.vertList.values())
+
+    def dijkstra(self, src, dest):
+
+        assert src in self.vertList, 'Such source node doesn\'t exist'
+
+        distances = {vertex: np.Infinity for vertex in self.vertList}
+        distances[src] = 0
+
+        previous_vertices = {
+            vertex: None for vertex in self.vertList
+        }
+
+        vertices = self.vertList.copy()
+
+        while vertices:
+
+            current_vertex = min(
+                vertices, key=lambda vertex: distances[vertex])
```

Comment:
If I'm reading this correctly, this takes linear work to fetch the min value from the array which  I believe means that dijkstra's algo becomes quadratic. Pretty sure you need a priority queue to properly implement the algo although my memory is pretty fuzzy at the moment.

Standard heap implementation should do the trick: https://docs.python.org/2/library/heapq.html